ddtrace/contrib/internal/azure_eventhubs/utils.py

```diff
@@ -0,0 +1,125 @@
+from typing import Union
+from uuid import UUID
+
+import azure.eventhub as azure_eventhub
+import azure.eventhub.amqp as azure_eventhub_amqp
+
+from ddtrace import config
+from ddtrace.contrib.trace_utils import ext_service
+from ddtrace.ext import SpanTypes
+from ddtrace.ext import azure_eventhubs as azure_eventhubsx
+from ddtrace.internal import core
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.propagation.http import HTTPPropagator
+
+
+def create_context(context_name, pin, operation_name, resource=None):
+    return core.context_with_data(
+        context_name,
+        span_name=operation_name,
+        pin=pin,
+        resource=resource,
+        service=ext_service(pin, config.azure_eventhubs),
+        span_type=SpanTypes.WORKER,
+    )
+
+
+def handle_event_hubs_event_data_context(span, event_data_arg_value):
+    if isinstance(event_data_arg_value, (azure_eventhub.EventData, azure_eventhub_amqp.AmqpAnnotatedMessage)):
+        inject_context(span, event_data_arg_value)
+    elif (
+        isinstance(event_data_arg_value, list)
+        and event_data_arg_value
+        and isinstance(event_data_arg_value[0], (azure_eventhub.EventData, azure_eventhub_amqp.AmqpAnnotatedMessage))
+    ):
+        for event in event_data_arg_value:
+            inject_context(span, event)
+    elif isinstance(event_data_arg_value, azure_eventhub.EventDataBatch):
+        for event_data in event_data_arg_value._internal_events:
+            parent_context = extract_context(event_data)
+            if (
+                parent_context is not None
+                and parent_context.trace_id is not None
+                and parent_context.span_id is not None
+            ):
+                span.link_span(parent_context)
+
+
+def extract_context(event_data):
+    msg = event_data if isinstance(event_data, azure_eventhub_amqp.AmqpAnnotatedMessage) else event_data._message
+    return HTTPPropagator.extract(msg.application_properties)
+
+
+def inject_context(span, event_data):
```

Comment:
is this used anywhere? I didn't see a reference to it (but GitHub is giving me a weird UI for reviewing since there are too many files in the PR 😅)

Reply:
It's used in two places in the same file:
- https://github.com/DataDog/dd-trace-py/blob/c5c621a77d7df37e7bf0281a512aa86d17d1dd23/ddtrace/contrib/internal/azure_eventhubs/utils.py#L45
- https://github.com/DataDog/dd-trace-py/blob/c5c621a77d7df37e7bf0281a512aa86d17d1dd23/ddtrace/contrib/internal/azure_eventhubs/utils.py#L52